lambdas/preview/index.py

```diff
@@ -141,35 +145,102 @@ def extract_vcf(file_):
     """
     Pull summary info from VCF: meta-information, header line, and data lines
     (in that order, up to MAX_LINES)
-    """
 
+    Args:
+        file_ - file-like
+    """
     meta = []
     header = []
     data = []
-
+    size = 0
     with open(file_.name, 'rb') as vcf:
         for index, line in enumerate(vcf, start=1):
-            # don't escape quotes
-            line = line.strip().decode('unicode_escape')
-            formatted = f'<p>{line}</p>'
+            line = _truncate(line.rstrip().decode(), size)
             if line.startswith('##'):
-                meta.append(formatted)
+                meta.append(line)
             elif line.startswith('#'):
-                header.append(formatted)
+                header.append(line)
             else:
-                data.append(formatted)
+                data.append(line)
+            size += len(line)
+            # stop if we're over the max
+            if index >= MAX_LINES or size > MAX_BYTES:
+                break
 
-            if index >= MAX_LINES:
+    info = {
+        'data': {
+            'meta': meta,
+            'header': header,
+            'data': data,
+        }
+    }
+
+    return '', info
+
+def extract_txt(file_):
+    """
+    Display first and last few lines of a potentially large file.
+
+    Args:
+        file_ - file-like object
+    """
+    size = 0
+    ellipsis = False
+    head = []
+    with open(file_.name, 'rb') as txt:
```

Comment:
Do you actually need to open it again? `file_` is already an open file.

Same for the VCF function.